lib/Backend/DecompileFunction.cpp

```diff
@@ -1690,16 +1648,15 @@ RecursiveCoroutine<void> CCodeGenerator::emitGHASTNode(const ASTNode *N) {
       if (isLocalVarDecl(VarDeclCall) or isCallStackArgumentDecl(VarDeclCall)) {
         std::string VarName = createLocalVarDeclName(VarDeclCall);
         revng_assert(not VarName.empty());
-        Out << getNamedCInstance(TypeMap.at(VarDeclCall), VarName, B) << ";\n";
+        Out << getNamedCInstance(*TypeMap.at(VarDeclCall), VarName, B) << ";\n";
```

Comment:
@Cvelth I was just thinking about this for something else I'm doing and something occurred to me.

As you know, at the moment in C we can't emit variable declarations with inline initialization. Not for some substantial problem, but we haven't implemented it yet. As a result if `TypeMap.at(VarDeclCall)` returns a `const`-qualified type we will end up generating C code that doesn't compile.

I know in other places we try and take precautions to avoid this, but at the moment there isn't any strong guarantee that `TypeMap.at(VarDeclCall)` can't return a `const`-qualified type.

I think that the most robust thing to do here is to detect if we have `const`-ness, and strip it away before passing the type as first argument to `getNamedCInstance`. In fact this is what I'm doing on another branch of mine to prevent emitting invalid C code. Please do this in this patchset too.

Reply:
wait, I'm pretty sure I have already seen something like this! Did you by any chance already implemented it (here or in some other `TypeMap`-adjacent place)?

Reply:
Great, thanks for double checking this.